docs/proposals/koordlet-architecture.md

```diff
@@ -0,0 +1,70 @@
+# Koordlet Design and Architecture
```

Comment:
Relocate this file under `docs/` instead of `docs/proposals/` folder might be better since we're already developing based on this architecture.

Reply:
mv to desgin-archive